platform-modules/batch-models/src/main/scala/org/ekstep/analytics/job/report/BaseCourseMetrics.scala

```diff
@@ -0,0 +1,41 @@
+package org.ekstep.analytics.job.report
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.{DataFrame, Encoders, SparkSession}
+import org.ekstep.analytics.framework._
+import org.ekstep.analytics.util.CourseUtils.{getCourseBatchDetails, getLiveCourses, getTenantInfo, loadData}
+import org.sunbird.cloud.storage.conf.AppConf
+import org.ekstep.analytics.framework.util.CommonUtil
+
+case class BaseCourseMetricsOutput(courseName: String, batchName: String, status: String, slug: String, courseId: String, batchId: String) extends AlgoInput
+
+trait BaseCourseMetrics[T <: AnyRef, A <: BaseCourseMetricsOutput, B <: AlgoOutput, R <: Output] extends IBatchModelTemplate[T,BaseCourseMetricsOutput,B,R]{
+
+  override def preProcess(events: RDD[T], config: Map[String, AnyRef])(implicit sc: SparkContext, fc: FrameworkContext): RDD[BaseCourseMetricsOutput] = {
```

Comment:
Configuration is incorrect. It should take in the following parameters
- channelId - Array[String]
- courseId - Array[String]
- esQuery - String (Elastic search query)

Reply:
Done.